### PR TITLE
fix graph break for torch.compile in attention mask check

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?

`torch.compile` gives a graph break:
```
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0] Graph break from Tensor.item(), consider setting:
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]     torch._dynamo.config.capture_scalar_outputs = True
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0] or:
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0] to include these operations in the captured graph.
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0] Graph break: from user code at:
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]   File "/root/miniconda3/envs/py3.11/lib/python3.11/site-packages/transformers/models/qwen3/modeling_qwen3.py", line 540, in torch_dynamo_resume_in_forward_at_526
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]     causal_mask = self._update_causal_mask(
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]   File "/root/miniconda3/envs/py3.11/lib/python3.11/site-packages/transformers/models/qwen3/modeling_qwen3.py", line 617, in updatecausal_mask
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]     if attention_mask is not None and 0.0 in attention_mask:
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]
W0503 22:06:01.020000 30493 site-packages/torch/_dynamo/variables/tensor.py:869] [12/0]
```

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by vasqu** [comment]
Just a bit unsure since a user had issues with a similar version in #37199 (first graph break they mention). Haven't had time yet too look into it. 

Which torch version are you using btw?
